IdentityServer/v6/docs/content/quickstarts/1_client_credentials.md

```diff
@@ -123,51 +147,126 @@ public static IEnumerable<Client> Clients =>
     };
 ```
 
-You can think of the ClientId and the ClientSecret as the login and password for your application itself.  
-It identifies your application to the identity server so that it knows which application is trying to connect to it.	
+Again, see the full file [here]({{< param qs_base >}}/1_ClientCredentials/src/IdentityServer/Config.cs).
+
+Clients can be configured with many options. Your minimal machine-to-machine
+client here contains 
+- A ClientId, which identifies the application to IdentityServer so that it
+  knows which client is trying to connect to it.
+- A Secret, which you can think of as the password for the client.
+- The list of scopes that the client is allowed to ask for.
+
+Notice that the allowed scope here matches the name of the ApiScope above.
 
-	
-#### Configuring IdentityServer
-Loading the resource and client definitions happens in [Startup.cs]({{< param qs_base >}}/1_ClientCredentials/src/IdentityServer/Startup.cs) - update the code to look like this
+### Configuring IdentityServer
+The scope and client definitions are loaded in 
+[HostingExtensions.cs]({{< param qs_base >}}/1_ClientCredentials/src/IdentityServer/HostingExtensions.cs). 
+The template created a ConfigureServices method there that is already loading
+the scopes and clients. You can take a look to see how it is done. Note that the
+template adds a few things that are not used in this quickstart. Here's the
+minimal ConfigureServices method that is needed:
 
 ```csharp
-public void ConfigureServices(IServiceCollection services)
+public static WebApplication ConfigureServices(this WebApplicationBuilder builder)
 {
     var builder = services.AddIdentityServer()
         .AddInMemoryApiScopes(Config.ApiScopes)
         .AddInMemoryClients(Config.Clients);
 
-    // omitted for brevity
+    return builder.Build();
 }
 ```
 
-That's it - your identity server should now be configured. If you run the server and navigate the browser to *https://localhost:5001/.well-known/openid-configuration*, you should see the so-called discovery document. 
-The discovery document is a standard endpoint in identity servers.  The discovery document will be used by your clients and APIs to download the necessary configuration data.
+That's it - your IdentityServer is now configured. If you run the project and
+then navigate to *https://localhost:5001/.well-known/openid-configuration* in
+your browser, you should see the [discovery document]({{< ref "/reference/endpoints/discovery">}}). 
+The discovery document is a standard endpoint in [OpenID Connect](https://openid.net/specs/openid-connect-discovery-1_0.html) and 
+[OAuth](https://self-issued.info/docs/draft-ietf-oauth-discovery-01.html). It is
```

Comment:
Here's the correct link for the OAuth discovery spec: https://datatracker.ietf.org/doc/html/rfc8414